spec/client_spec.js

```diff
@@ -275,22 +275,50 @@ describe('Client', function() {
         requestsCount++;
       });
 
-      it('returns a promise', function() {
+      it('returns a promise', function(done) {
         promise = subject.get('ticket.subject');
 
         expect(promise).to.be.a.promise;
+        expect(promise).to.eventually.become('test').and.notify(done);
+
+        window.addEventListener.callArgWith(1, {
+          origin: subject._origin,
+          source: subject._source,
+          data: { id: requestsCount, result: { errors: {}, 'ticket.subject': 'test' } }
+        });
       });
 
-      it('accepts an array with multiple paths', function() {
-        promise = subject.get(['ticket.subject', 'ticket.requester']);
+      it('throws an error when the handler throws it', function(done) {
+        promise = subject.get('ticket.err');
 
         expect(promise).to.be.a.promise;
+        expect(promise).to.be.rejectedWith(Error, 'ticket.e').and.notify(done);
+
+        window.addEventListener.callArgWith(1, {
+          origin: subject._origin,
+          source: subject._source,
+          data: { id: requestsCount, result: { errors: { 'ticket.err': { message: 'ticket.err unavailable' } } } }
+        });
       });
 
-      it('accepts multiple arguments', function() {
-        promise = subject.get('ticket.subject', 'ticket.requester');
+      it('accepts an array with multiple paths', function(done) {
+        promise = subject.get(['ticket.subject', 'ticket.requester']);
 
         expect(promise).to.be.a.promise;
+        expect(promise).to.eventually.become({ 'ticket.subject': 'test', 'ticket.requester': 'test' }).and.notify(done);
+
+        window.addEventListener.callArgWith(1, {
+          origin: subject._origin,
+          source: subject._source,
+          data: { id: requestsCount, result: { 'ticket.subject': 'test', 'ticket.requester': 'test' } }
+        });
+      });
+
+      it('doesn\'t accepts multiple arguments', function() {
+        requestsCount--;
```

Comment:
this requestsCount is pretty hacky - but we can solve that later in a different PR